p2p/unicast/router.go

```diff
@@ -0,0 +1,136 @@
+package unicast
+
+import (
+	"io"
+	"time"
+
+	libp2pHost "github.com/libp2p/go-libp2p/core/host"
+	libp2pNetwork "github.com/libp2p/go-libp2p/core/network"
+
+	"github.com/pokt-network/pocket/p2p/config"
+	typesP2P "github.com/pokt-network/pocket/p2p/types"
+	"github.com/pokt-network/pocket/p2p/utils"
+	"github.com/pokt-network/pocket/shared/modules"
+	"github.com/pokt-network/pocket/shared/modules/base_modules"
+)
+
+// TECHDEBT(#629): configure timeouts. Consider security exposure vs. real-world conditions.
+// TECHDEBT(#629): parameterize and expose via config.
+// readStreamTimeout is the duration to wait for a read operation on a
+// stream to complete, after which the stream is closed ("timed out").
+const readStreamTimeout = time.Second * 10
+
+var _ unicastRouterFactory = &UnicastRouter{}
+
+type unicastRouterFactory = modules.FactoryWithConfig[*UnicastRouter, *config.UnicastRouterConfig]
+
+type UnicastRouter struct {
+	base_modules.IntegratableModule
+
+	logger *modules.Logger
+	host   libp2pHost.Host
+	// messageHandler is the function to call when a message is received.
+	// host represents a libp2p network node, it encapsulates a libp2p peerstore
+	// & connection manager. `libp2p.New` configures and starts listening
+	// according to options.
+	// (see: https://pkg.go.dev/github.com/libp2p/go-libp2p#section-readme)
+	messageHandler typesP2P.MessageHandler
+	peerHandler    func(peer typesP2P.Peer) error
+}
+
+func Create(bus modules.Bus, cfg *config.UnicastRouterConfig) (*UnicastRouter, error) {
+	return new(UnicastRouter).Create(bus, cfg)
+}
+
+func (*UnicastRouter) Create(bus modules.Bus, cfg *config.UnicastRouterConfig) (*UnicastRouter, error) {
```

Comment:
```suggestion
// All fields in the cfg are required
func (*UnicastRouter) Create(bus modules.Bus, cfg *config.UnicastRouterConfig) (*UnicastRouter, error) {
```

Reply:
Do you feel strongly about adding this comment? After the submodule formalization and subsequent documentation updates it will be conventional that configs contain required values by definition. Additionally, these configs implement an `#IsValid()` method which may be enforced by an interface in the future.

Reply:
Perhaps an ADR around the module system would be appropriate?

Reply:
Don't feel strongly about this. Resolving